src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/UserRegistrationController.kt

```diff
@@ -47,19 +47,19 @@ class UserRegistrationController(
             return "registration/registration_disabled"
         }
 
-        if (!user.userName.isEmpty() && userRepository.findByUserName(user.userName) != null) {
+        if (!user.userName.isBlank() && userRepository.findByUserName(user.userName) != null) {
             result.rejectValue("userName", "username.exists", "username unavailable")
         }
 
-        if (!user.email.isEmpty() && userRepository.findByEmail(user.email) != null) {
+        if (!user.email.isBlank() && userRepository.findByEmail(user.email) != null) {
             result.rejectValue("email", "email.exists", "email already in use")
         }
 
         if (result.hasErrors()) {
             return "registration/create_account"
         }
 
-        val newUser = User(null, user.userName, null, user.email, user.firstName, user.lastName, false)
+        val newUser = User(null, user.userName, null, user.email, user.firstName, user.lastName, true)
```

Comment:
Constructor without nulls would be better.

Reply:
fixed